lib/post_destroyer.rb

```diff
@@ -90,6 +90,8 @@ def destroy
       UserActionManager.topic_destroyed(@topic)
       DiscourseEvent.trigger(:topic_destroyed, @topic, @user)
       WebHook.enqueue_topic_hooks(:topic_destroyed, @topic, topic_payload) if has_topic_web_hooks
+      Discourse.reset_tos_path! if SiteSetting.tos_topic_id == @topic.id
+      Discourse.reset_privacy_policy_path! if SiteSetting.privacy_topic_id == @topic.id
```

Comment:
I think we can avoid all this reset code by just including the `id`s in the cache key. `Discourse.cache` has a default expiry of 1 day by default so we don't have to worry about clean up.